### ISSUE TITLE
Internet Hosting Tool: After Windows Sleep, unavailable from the outside

### ISSUE THREAD

**Issue by idc77** [moonlight-stream/nvidia-gamestream-issues#47]
Not sure where to post this, because the Internet Hosting Tool doesn't have an enabled issue tracker.

When I put Win10 to sleep, power it on again via IPMI, the system isn't reachable from the outside.
The IHT services need to be restarted.

But gamestream is available internally, if I connect via wireguard and then do a connection to gamestream.

**Comment by cgutman** [comment]
> Not sure where to post this, because the Internet Hosting Tool doesn't have an enabled issue tracker.

Here is fine I guess. I closed the issue tracker on that repo because it devolved entirely into support requests for router issues instead of actual bugs in the tool.

> When I put Win10 to sleep, power it on again via IPMI, the system isn't reachable from the outside.
The IHT services need to be restarted.

Interesting, I assume you are running the latest v5.6.1 version? I've heard of folks having issues with GameStream itself coming back online after sleep but not the Internet Hosting Tool services.

When you resume your system from sleep, are miss.exe and gsv6fwd.exe still running? Are there any messages in the Application or System event logs that indicate either service may have crashed?

The tool should refresh the port mappings about every 2 minutes. Can you check in your `C:\ProgramData\MISS` folder to see if the log files are still updating after you resume the machine? They should update immediately once your PC wakes up (due to the network change), but it may take 2 minutes if that's not working for some reason.

**Comment by idc77** [comment]
> When you resume your system from sleep, are miss.exe and gsv6fwd.exe still running? Are there any messages in the Application or System event logs that indicate either service may have crashed?

They are present in task manager. The services are running.

Not sure where to access those logs, I'll find out, next time I'm waking it from sleep.
It was up longer than 2 min without change.
I checked ProgramData/MISS/ , but logging could use timestamps.
They rotated 3 days after this issue was created, and all I could find was something was UNAUTHORIZED.

Maybe interesting, Windows wake up behavior is inconsistent.
1st time I woke it up, it was stuck at the login screen, Teamviewer however ran so I was able to log in.
2nd time there was no login screen ( I managed to see the desktop with moonlight over wireguard) but Teamviewer couldn't connect. Then gamestream locked up and I wasn't able to connect with Moonlight. I had to power cycle the computer (via IPMI).

I'll post more if I have more accurate information.

I also received an email saying I'm blocked from moonlight-stream on github on sept 26th 2023. Maybe by mistake, not sure what I did that's banworthy?

**Comment by cgutman** [comment]
> I checked ProgramData/MISS/ , but logging could use timestamps.

Yeah, the last modified time of the log itself is most relevant, since those logs are rotated every 2 minutes during normal operations.

> They rotated 3 days after this issue was created, and all I could find was something was UNAUTHORIZED.

Hmm, unauthorized usually means there are existing port forwarding rules for another address. If your router is assigning your PC a new IP address after it wakes up, that could explain the issue (though AFAIK the UPnP service is supposed to communicate with the DHCP service to kill port mappings for hosts that have lost their IP address lease). You could check your router settings and see if there’s an option for DHCP lease time and try extending that further, or assign your PC a static IP address if your router is constantly changing it.

> I also received an email saying I'm blocked from moonlight-stream on github on sept 26th 2023. Maybe by mistake, not sure what I did that's banworthy?

No idea. I definitely didn’t ban you. You must be unbanned now because I don’t think you could post here if you were banned.
